core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2459,6 +2493,40 @@ public static int minus(int b0, int b1) {
     throw notArithmetic("-", b0, b1);
   }
 
+  // checked -
+
+  public static byte checkedMinus(byte b0, byte b1) {
+    return intToByte(b0 - b1);
+  }
+
+  public static short checkedMinus(short b0, short b1) {
+    return intToShort(b0 - b1);
+  }
+
+  public static int checkedMinus(int b0, int b1) {
+    return Math.subtractExact(b0, b1);
+  }
+
+  public static long checkedMinus(long b0, long b1) {
+    return Math.subtractExact(b0, b1);
+  }
+
+  public static byte checkedUnaryMinus(byte b) {
+    return intToByte(-b);
+  }
+
+  public static short checkedUnaryMinus(short b) {
+    return intToShort(b);
```

Comment:
Why not `intToShort(-b)` here?

Reply:
I mean here it should be `intToShort(-b)` not `intToShort(b)`

Reply:
I have pushed a new commit with a fix and a test.